src/main/java/com/commercetools/sync/commons/utils/CommonTypeUpdateActionUtils.java

```diff
@@ -71,19 +72,28 @@
      *
      * @param oldResourceIdentifier the old resource identifier
      * @param newResourceIdentifier the new resource identifier
+     * @param useKeys               if true, the keys of the identifier are used for the comparison
      * @param <T>                   the type of the old resource identifier
      * @param <S>                   the type of the new resource identifier
      * @return true or false depending if the resource identifiers have the same id.
      */
     public static <T extends ResourceIdentifier, S extends ResourceIdentifier> boolean areResourceIdentifiersEqual(
-        @Nullable final T oldResourceIdentifier, @Nullable final S newResourceIdentifier) {
+        @Nullable final T oldResourceIdentifier,
+        @Nullable final S newResourceIdentifier,
+        boolean useKeys) {
```

Comment:
I would add a new method for this instead of passing a boolean check. So we could remove the old method when we don't need reference resolution after all resources support ResourceIdentifiers. It's also a smell in here, I could not see a reason to use a generic static utility method to get the key of the ResourceIdentifier with the method getKeyOfResourceIdentifier before we don't know the exact type but now we know the type. Am I wrong ? 🤔

Reply:
it doesn`t smell because ,we don`t now if the given Resource identifier is a Reference. if yes, we have to fetch its key from its objects, otherwise we can get the key from the resource identifier directly

Reply:
I would leave the boolean check in the method, because right now we can use the same method, for key-  and id comparison. And when we have finished the refactoring of the resolutionresolving, we can simply remove the boolean check than.